syntax = "proto3";

package trading;

import "trading/transformer.proto";

// PersistenceService saves transformed quotes to the database.
service PersistenceService {
  // Persist a single quote to the database.
  rpc Persist(PersistRequest) returns (PersistResponse);
  
  // Persist multiple quotes to the database.
  rpc BulkPersist(BulkPersistRequest) returns (BulkPersistResponse);
}

message PersistRequest {
  TransformResponse quote = 1;
}

message PersistResponse {
  bool success = 1;
  string symbol = 2;
  string message = 3;
}

message BulkPersistRequest {
  repeated TransformResponse quotes = 1;
}

message BulkPersistResponse {
  repeated PersistResponse results = 1;
}
